glib/src/utils.rs

```diff
@@ -244,16 +258,19 @@ mod tests {
         );
         assert_eq!(crate::uri_parse_scheme("foo"), None);
 
-        let escaped = crate::uri_escape_string("&foo", None, true);
+        let escaped = crate::uri_escape_string("&foo", crate::NONE_STR, true);
         assert_eq!(escaped, GString::from("%26foo"));
 
-        let unescaped = crate::uri_unescape_string(escaped.as_str(), None);
+        let unescaped = crate::uri_unescape_string(escaped.as_str(), crate::GStr::NONE);
         assert_eq!(unescaped, Some(GString::from("&foo")));
 
         assert_eq!(
-            crate::uri_unescape_segment(Some("/foo"), None, None),
+            crate::uri_unescape_segment(Some("/foo"), crate::NONE_STR, crate::NONE_STR),
             Some(GString::from("/foo"))
         );
-        assert_eq!(crate::uri_unescape_segment(Some("/foo%"), None, None), None);
+        assert_eq!(
+            crate::uri_unescape_segment(Some("/foo%"), crate::NONE_STR, crate::NONE_STR),
```

Comment:
why do we provide both `NONE_STR` & `GStr::NONE`?

Reply:
Just for consistency. `NONE_STR` is not really necessary

Reply:
I think we should move all these `NONE_BLA` constants into `Bla::NONE` for better namespacing, but that's something for a separate PR.